### PR TITLE
refactor(streaming):  merge OrderedRowSerializer and OrderedRowDeserializer into one

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

`OrderedRowSerializer` and `OrderedRowDeserializer` share identical fields and we can implement them as one structure,  the serializer need to know the schema info. 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/issues/5278

### CODE REVIEW

File: src/stream/src/executor/managed_state/aggregation/mod.rs [hunk 1]
```diff
@@ -185,6 +187,7 @@ impl<S: StateStore> ManagedStateImpl<S> {
         pk_indices: &PkIndices,
         group_key: Option<&Row>,
         extreme_cache_size: usize,
+        input_schema: Schema,
```
Comment (richardchien) [inline src/stream/src/executor/managed_state/aggregation/mod.rs:190]:
```suggestion
        input_schema: &Schema,
```

File: src/stream/src/executor/managed_state/aggregation/mod.rs [hunk 2]
```diff
@@ -204,6 +207,7 @@ impl<S: StateStore> ManagedStateImpl<S> {
                     .clone(),
                 row_count,
                 extreme_cache_size,
+                input_schema,
```
Comment (richardchien) [inline src/stream/src/executor/managed_state/aggregation/mod.rs:210]:
```suggestion
                input_schema.clone(),
```

File: src/stream/src/executor/managed_state/aggregation/extreme.rs [hunk 3]
```diff
@@ -107,14 +108,17 @@ impl<S: StateStore> GenericExtremeState<S> {
...
         let upstream_agg_col_idx = agg_call.args.val_indices()[0];
         // map agg column to state table column index
         let state_table_agg_col_idx = col_mapping
             .upstream_to_state_table(agg_call.args.val_indices()[0])
             .expect("the column to be aggregate must appear in the state table");
         // map order by columns to state table column indices
-        let (state_table_order_col_indices, state_table_order_types) = std::iter::once((
+        let state_table_order_col_indices: Vec<usize>;
+        let state_table_order_types: Vec<OrderType>;
+        (state_table_order_col_indices, state_table_order_types) = std::iter::once((
```
Comment (richardchien) [inline src/stream/src/executor/managed_state/aggregation/extreme.rs:121]:
```suggestion
        let (state_table_order_col_indices, state_table_order_types): (Vec<_>, Vec<_>) = std::iter::once((
```

File: src/stream/src/executor/hash_agg.rs [hunk 4]
```diff
@@ -300,6 +301,7 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
                                     result_table,
                                     input_pk_indices,
                                     *extreme_cache_size,
+                                    input_schema.clone(),
```
Comment (richardchien) [inline src/stream/src/executor/hash_agg.rs:304]:
```suggestion
                                    input_schema,
```

File: src/stream/src/executor/global_simple_agg.rs [hunk 5]
```diff
@@ -164,6 +164,7 @@ impl<S: StateStore> GlobalSimpleAggExecutor<S> {
                     result_table,
                     input_pk_indices,
                     extreme_cache_size,
+                    input_schema.clone(),
```
Comment (richardchien) [inline src/stream/src/executor/global_simple_agg.rs:167]:
```suggestion
                    input_schema,
```

File: src/stream/src/executor/aggregation/mod.rs [hunk 6]
```diff
@@ -282,6 +283,7 @@ pub async fn generate_managed_agg_state<S: StateStore>(
     result_table: &StateTable<S>,
     pk_indices: &PkIndices,
     extreme_cache_size: usize,
+    input_schema: Schema,
```
Comment (richardchien) [inline src/stream/src/executor/aggregation/mod.rs:286]:
```suggestion
    input_schema: &Schema,
```

File: src/stream/src/executor/aggregation/mod.rs [hunk 7]
```diff
@@ -317,6 +318,7 @@ pub async fn generate_managed_agg_state<S: StateStore>(
                 pk_indices,
                 group_key.as_ref(),
                 extreme_cache_size,
+                input_schema.clone(),
```
Comment (richardchien) [inline src/stream/src/executor/aggregation/mod.rs:321]:
```suggestion
                input_schema,
```

**Review by richardchien — COMMENTED** [review]
Suggested some changes to avoid unnecessary cloning. Rest part of streaming aggregation LGTM!

**Review by st1page — APPROVED** [review]
LGTM

**Pull request merged**